src/Microsoft.VisualStudio.Threading/AsyncSemaphore.cs

```diff
@@ -24,21 +30,29 @@ public class AsyncSemaphore : IDisposable
         private static readonly Task<Releaser> DisposedReleaserTask = TplExtensions.FaultedTask<Releaser>(new ObjectDisposedException(typeof(AsyncSemaphore).FullName));
 
         /// <summary>
-        /// The semaphore used to keep concurrent access to this lock to just 1.
+        /// A task that is canceled without a specific token.
         /// </summary>
-#pragma warning disable CA2213 // Disposable fields should be disposed
-        private readonly SemaphoreSlim semaphore;
-#pragma warning restore CA2213 // Disposable fields should be disposed
+        private static readonly Task<Releaser> CanceledReleaser = Task.FromCanceled<Releaser>(new CancellationToken(true));
 
         /// <summary>
         /// A task to return for any uncontested request for the lock.
         /// </summary>
         private readonly Task<Releaser> uncontestedReleaser;
 
         /// <summary>
-        /// A task that is cancelled.
+        /// The sync object to lock on for sensitive operations.
```

Comment:
What is a "sensitive operation"?

Reply:
Field access. I can clarify in the code comment.